Makefile

```diff
@@ -16,10 +16,13 @@ VSCODE_TESTS_BASE_URL = 'https://raw.githubusercontent.com/microsoft/vscode/e6a4
 build: generate build-quick
 
 build-quick:
-	go build -trimpath -ldflags "-s -w $(GOVARS) $(ADDITIONAL_GO_LINKER_FLAGS)" ./cmd/micro
+	CGO_ENABLED=0 go build -trimpath -ldflags "-s -w $(GOVARS) $(ADDITIONAL_GO_LINKER_FLAGS)" ./cmd/micro
+
+build-dynamic: generate
+	CGO_ENABLED=1 go build -trimpath -ldflags "-s -w $(GOVARS) $(ADDITIONAL_GO_LINKER_FLAGS)" ./cmd/micro
```

Comment:
Seems a bit messy, and "asymmetric" (no quick version of `build-dynamic`, no dynamic version of `build-dbg`). Maybe instead of adding a target, just add a variable, e.g. `DYNAMIC`? I.e. just:

```
build-quick:
	CGO_ENABLED=$(DYNAMIC) go build ...

build-dbg:
	CGO_ENABLED=$(DYNAMIC) go build ...
```

And then we can run `DYNAMIC=1 make build` or whatever.

Also need to update (i.e. rewrite) the `Fully static binary` section in README.

Reply:
Damn, this "hidden" `darwin` dependency `tools/info-plist.go` :man_facepalming: 
So we've to exclude `darwin` somehow or can't generalize it in the Makefile.

Maybe define `ADDITIONAL_GO_LINKER_FLAGS` to be empty and extend it only in case the `GOOS` is `darwin` and forcing `CGO_ENABLED` to be `1` afterwards. :thinking:

Reply:
Yeah, my first thought was: what the hell is this plist and can we just get rid of it or its `-linkmode external`. But it is not obvious if it is possible and how, and I have no wish to figure that out.

> Maybe define `ADDITIONAL_GO_LINKER_FLAGS` to be empty and extend it only in case the `GOOS` is `darwin` and forcing `CGO_ENABLED` to be `1` afterwards. 🤔

Seems to make sense.

Reply:
BTW found its roots: #513

Reply:
> Maybe define `ADDITIONAL_GO_LINKER_FLAGS` to be empty and extend it only in case the `GOOS` is `darwin` and forcing `CGO_ENABLED` to be `1` afterwards. 🤔

Should do the trick (for) now.
But to be honest, this kind of meta info is only included in the moment `micro` is natively build on macOS. It's completely missing, at least since the builds are cross compiled via GitHub actions, in our prebuilt binaries. :man_shrugging:

Reply:
Yeah, I've *just* realized that, after looking at GOHOSTOS in your patch.

Which makes this feature from #513 even more... interesting.

Reply:
Although before https://github.com/zyedidia/micro/commit/6945aa34eb39e4ad8698dc1312fe72740b074a9f this feature was used for cross-compiled binaries as well... but I doubt it was really working correctly.

Reply:
Now, should we document this (the fact that MacOS builds are an exception, but only native ones) in README as well?

Reply:
Yep, better to document before someone is complaining.

> Although before https://github.com/zyedidia/micro/commit/6945aa34eb39e4ad8698dc1312fe72740b074a9f this feature was used for cross-compiled binaries as well... but I doubt it was really working correctly.

Me too, since building `micro` with `darwin` AND `CGO_ENABLED=1` requires `clang`.
We could try it with `GOOS` instead of `GOHOSTOS` as switch, but then we most probably run into trouble cross compiling for `darwin` where `clang` is not available. This might hit the GitHub actions etc.pp